### DISCUSSION TITLE
arguments --assign and --user_name

### DISCUSSION THREAD

**Discussion by williamharwell** [rustdesk/rustdesk#5734] [Q&A]
How are the arguments --assign and --user_name used? Is it possible to assign a client to a user, in the pro version,  while using an install script?

**Comment by rustdesk** [comment]
@dinger1986

**Reply by dinger1986** [reply]
shall take a look and document it as well

**Comment by dinger1986** [comment]
> How are the arguments --assign and --user_name used? Is it possible to assign a client to a user, in the pro version, while using an install script?

Do you mean auto sign in a user?

**Comment by williamharwell** [comment]
I guess I misunderstood what the login function does. I’m actually looking for a way to prevent someone that does not have an account on the server from accessing a device.




How are the arguments --assign and --user_name used? Is it possible to assign a client to a user, in the pro version, while using an install script?

Do you mean auto sign in a user?

**Reply by dinger1986** [reply]
You want to prevent a specific person from accessing a specific machine or you want to prevent anyone not signed in from using your server to connect to other devices?

**Reply by williamharwell** [reply]
Sorry, I DO mean auto sign in a user.

**Reply by dinger1986** [reply]
From my tests with that command it assigns it in the API but doesn't log in on the device

**Reply by williamharwell** [reply]
That's done during setup anyway if the network fields are populated.

**Reply by dinger1986** [reply]
No sorry if you do `.\rustdesk.exe --assign --token "generatedtoken" --user_name "testusera"` it changes the user its assigned to on the web console but doesnt sign the client in, it needs the API settings set first in the Client Network settings same with Strategy

**Reply by williamharwell** [reply]
Ah, that may be good enough since I have users in groups, and do not have an enabled default group. 
I have a user in group 1. Mainuser1
I assign the device to the user Mainuser1, which by default puts the device in the same group as Mainuser1.
Only other users in Group1 can access the device. 
If an employee leaves or is terminated, I remove their account, thus ensuring they do not have access to the device, even if they have the network information and the permanent password.

Am I missing anything? Can this be circumvented?

**Reply by dinger1986** [reply]
That is 100% correct, sorry think I was maybe just misunderstanding what you wanted to do.

So you could do assign in group b which everyone is in and then either delete the user or on the web console edit the device and simply remove the user from the device then it wouldn't have rights to access and no way to circumvent this.
